### PR TITLE
fix(graphql-auth-transformer): add list support for ownerField in subs

### PR DESCRIPTION
**Pull request by SwaySway**
support list ownerFields in subscriptions

re #3148
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/graphql-auth-transformer/src/ModelAuthTransformer.ts [hunk 1]
```diff
@@ -1784,15 +1786,28 @@ All @auth directives used on field definitions are performed when the field is r
     ctx.mapResourceToStack(parent.name.value, resolverResourceId);
   }
 
-  private addSubscriptionOwnerArgument(ctx: TransformerContext, resolver: Resolver, ownerRules: AuthRule[], makeNonNull: boolean = false) {
+  private addSubscriptionOwnerArgument(
+    ctx: TransformerContext,
+    resolver: Resolver,
+    currentFields: any,
```
Comment (yuth) [inline packages/graphql-auth-transformer/src/ModelAuthTransformer.ts]: Why any? From what I can see it should be of type `Record<string,string>`

File: packages/graphql-auth-transformer/src/ModelAuthTransformer.ts [hunk 2]
```diff
@@ -1784,15 +1786,28 @@ All @auth directives used on field definitions are performed when the field is r
...
       field => field.name.value === resolver.Properties.FieldName
     ) as FieldDefinitionNode;
-    const nameNode: any = makeNonNull ? makeNonNullType(makeNamedType('String')) : makeNamedType('String');
-    // const createArguments = [makeInputValueDefinition(DEFAULT_OWNER_FIELD, nameNode)];
     const ownerArgumentList = ownerRules.map(rule => {
-      return makeInputValueDefinition(rule.ownerField || DEFAULT_OWNER_FIELD, nameNode);
+      let argNode: any;
+      // if ownerField is defined read from fields if the value exists
+      if (rule.ownerField) {
+        argNode = currentFields[rule.ownerField] ? unwrapNonNull(currentFields[rule.ownerField]) : makeNamedType('String');
```
Comment (yuth) [inline packages/graphql-auth-transformer/src/ModelAuthTransformer.ts]: when would this be false?

Comment (SwaySway) [reply packages/graphql-auth-transformer/src/ModelAuthTransformer.ts]: This happens in the event that a ownerField rule is defined and the field is not added such as adding an ownerField named "notAllowed".

File: packages/graphql-transformer-core/src/util/getFieldArguments.ts [hunk 3]
```diff
@@ -4,12 +4,12 @@ import { FieldDefinitionNode } from 'graphql';
  * Given a Type returns a plain JS map of its arguments
  * @param arguments The list of argument nodes to reduce.
  */
-export function getFieldArguments(type: any): any {
+export function getFieldArguments(type: any, withBaseType: boolean = true): any {
```
Comment (yuth) [inline packages/graphql-transformer-core/src/util/getFieldArguments.ts]: can `any` to appropriate types

**Pull request merged**